src/cvmodel_package/src/cvmodel_node.cpp

```diff
@@ -0,0 +1,191 @@
+#include <cstdio>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <cv_bridge/cv_bridge.h>
+#include <opencv2/opencv.hpp>
+
+using std::placeholders::_1;
+using namespace cv;
+using namespace std;
+
+double sim(double a, double b);
+double distSq(Point2f a, Point2f b);
+double angle(Point2f a, Point2f b);
+vector<vector<Point>> getContours(Mat& frame, const string& color);
+Mat drawCenters(Mat frame, const string& color);
+
+Mat applyCanny(Mat& frame) {
+    Mat edges;
+    GaussianBlur(frame, frame, Size(5, 5), 1.5);
+    // Canny with thresholds 100 and 200
+    Canny(frame, edges, 100, 200);
+    Mat edgesColor;
+    cvtColor(edges, edgesColor, COLOR_GRAY2BGR);
+    return edgesColor;
+}
+
+class CVNode : public rclcpp::Node {
+public:
+    CVNode() : Node("CVNode"), writer_initialized(false), last_frame_time(this->now()) {
+        printf("Initializing subscriber...\n");
+        subscription_ = this->create_subscription<sensor_msgs::msg::Image>(
+            "image_raw", 10, std::bind(&CVNode::topic_callback, this, _1));
+
+        timer_ = this->create_wall_timer(
+            std::chrono::seconds(2), std::bind(&CVNode::check_for_inactivity, this));
+    }
+
+private:
+    VideoWriter writer;
+    bool writer_initialized;
+    std::string output_video = "output.mp4";
+    rclcpp::Time last_frame_time;
+    rclcpp::Subscription<sensor_msgs::msg::Image>::SharedPtr subscription_;
+    rclcpp::TimerBase::SharedPtr timer_;
+
+    void topic_callback(const sensor_msgs::msg::Image &msg) {
+        try {
+            cv_bridge::CvImagePtr cv_ptr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8);
+            Mat frame = cv_ptr->image;
+
+            Mat edges = applyCanny(frame);
+
+            frame = drawCenters(frame, "blue");
+
+            // Combine canny with center image procssing
+            // frame = frame & edges;
+
+            if (!writer_initialized) {
+                int frame_width = frame.cols;
+                int frame_height = frame.rows;
+                double fps = 120.0;
+
+                writer.open(output_video, VideoWriter::fourcc('X', 'V', 'I', 'D'), fps, Size(frame_width, frame_height));
```

Comment:
add in a parameter for video writer. in the standard use case, we don't want to output to video